### PR TITLE
Move `PlatformSpecificWindowBuilderAttributes` into `WindowAttributes`

### LINKED ISSUE #3313: Platform-specific getters on `WindowAttributes`
Currently `WindowAttributes`s purpose is to expose the current state of `WindowBuilder`. But it doesn't support platform-specific fields.
It would be nice to move `PlatformSpecificWindowBuilderAttributes` into `WindowAttributes` so we can do that.

Alternatively, I'm not entirely sure why `WindowAttributes` exists in the first place if seemingly we could let users just get free (mutable) access to `WindowBuilder` fields and the ones we don't want them to get access to can have getters/setters.

What do you guys think?

### PR DESCRIPTION
**Pull request by daxpedda**
Fixes #3313.
Any unsoundness this expands on is addressed by #3303.

This doesn't actually implement any getters, these can be implemented in follow-ups as needed.

### CODE REVIEW

**Review by madsmtm — APPROVED** [review]
I've been hesitant to review this because I dislike `WindowAttributes` on a philosophical level, but I think this is the right approach forwards.

Would also make it easier to move towards `event_loop.create_window(attributes)` as proposed in https://github.com/rust-windowing/winit/issues/3367, which would in turn make https://github.com/rust-windowing/winit/pull/3299 easier to move forwards on.

**Comment by kchibisov** [comment]
>I've been hesitant to review this because I dislike WindowAttributes on a philosophical level, but I think this is the right approach forwards.

In general the `WindowAttributes` and the builder in its current form will be gone in 0.31, because you can't extend builder when you own the type, and we'll have the trait for stuff like that.

File: src/window.rs [hunk 1]
```diff
@@ -159,6 +156,9 @@ pub struct WindowAttributes {
     #[cfg(feature = "rwh_06")]
     pub(crate) parent_window: Option<SendSyncRawWindowHandle>,
     pub fullscreen: Option<Fullscreen>,
+    // Platform-specific configuration.
+    #[allow(dead_code)]
+    pub(crate) platform_specific: platform_impl::PlatformSpecificWindowBuilderAttributes,
```
[the remark this answers is not in the document]
Comment (daxpedda) [reply src/window.rs:161]: I'm not sure what you mean.

[the remark this answers is not in the document]
Comment (kchibisov) [reply src/window.rs:161]: I wanted to say do `platform_specific: PlatformSpecificWindowBuilderAttributes`.

[the remark this answers is not in the document]
Comment (daxpedda) [reply src/window.rs]: Done.

**Pull request merged**